driver-core/src/test/unit/com/mongodb/AuthorizationHeaderTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb;
+
+import com.mongodb.internal.connection.AuthorizationHeader;
+import org.bson.internal.Base64;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.security.sasl.SaslException;
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class AuthorizationHeaderTest {
```

Comment:
The expected values for almost all of the tests were taken from the examples in https://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html.

Reply:
I would add two test cases
1. with `x-mongodb-gs2-cb-flag` and `x-mongodb-server-nonce`
2. with `x-mongodb-gs2-cb-flag`, `x-mongodb-server-nonce` and `x-amz-security-token`

Reply:
I added a test in testCreateCanonicalRequest that additionally adds all 3 of the values in the second case and ensures that they were added to the canonical request. I'm not sure what would be gained by adding another test case that does not have the x-amz-security-token or if I'm missing what you meant.